src/jltypes.c

```diff
@@ -3244,21 +3244,26 @@ void jl_init_types(void)
                                         jl_any_type, jl_null, 32);
 
     tv = jl_tuple1(tvar("T"));
+    jl_ref_type =
```

Comment:
GC is already disabled at this point in the init process. I guess we could assert at the top of the function that GC is disabled.